targets/csharp/templates/PlayFab_InstanceAPI.cs.ejs

```diff
@@ -0,0 +1,139 @@
+
+        public PlayFab<%- api.name %>InstanceAPI(PlayFabApiSettings settings)
+        {
+            apiSettings = settings;
+            authenticationContext = new PlayFabAuthenticationContext();
+        }
+
+        public PlayFab<%- api.name %>InstanceAPI(PlayFabAuthenticationContext context)
+        {
+            authenticationContext = context ?? new PlayFabAuthenticationContext();
+        }
+
+        public PlayFab<%- api.name %>InstanceAPI(PlayFabApiSettings settings, PlayFabAuthenticationContext context)
+        {
+            apiSettings = settings;
+            authenticationContext = context ?? new PlayFabAuthenticationContext();
+        }
+<% } else { %>
+        public PlayFab<%- api.name %>InstanceAPI(PlayFabAuthenticationContext context)
+        {
+            if (context == null)
+                throw new PlayFabException(PlayFabExceptionCode.AuthContextRequired, "Context cannot be null, create a PlayFabAuthenticationContext for each player in advance, or get <PlayFabClientInstanceAPI>.authenticationContext");
+            authenticationContext = context;
+        }
+
+        public PlayFab<%- api.name %>InstanceAPI(PlayFabApiSettings settings, PlayFabAuthenticationContext context)
+        {
+            if (context == null)
+                throw new PlayFabException(PlayFabExceptionCode.AuthContextRequired, "Context cannot be null, create a PlayFabAuthenticationContext for each player in advance, or get <PlayFabClientInstanceAPI>.authenticationContext");
+            apiSettings = settings;
+            authenticationContext = context;
+        }
+<% } %><% if (hasClientOptions) { %>
+        /// <summary>
+        /// Verify client login.
+        /// </summary>
+        public bool IsClientLoggedIn()
+        {
+            return authenticationContext == null ? false : authenticationContext.IsClientLoggedIn();
+        }
+<% } %><% if (hasEntityTokenOptions) { %>
+        /// <summary>
+        /// Verify entity login.
+        /// </summary>
+        public bool IsEntityLoggedIn()
+        {
+            return authenticationContext == null ? false : authenticationContext.IsEntityLoggedIn();
+        }
+<% } %>
+        /// <summary>
+        /// Clear the Client SessionToken which allows this Client to call API calls requiring login.
+        /// A new/fresh login will be required after calling this.
+        /// </summary>
+        public void ForgetAllCredentials()
+        {
+            authenticationContext?.ForgetAllCredentials();
+        }
+
+<% for(var i in api.calls) { var apiCall = api.calls[i];
+%><%- generateApiSummary("        ", apiCall, "summary")
+%><%- getDeprecationAttribute("        ", apiCall)
+%>        public async Task<PlayFabResult<<%- apiCall.result %>>> <%- apiCall.name %>Async(<%- apiCall.request %> request, object customData = null, Dictionary<string, string> extraHeaders = null)
+        {
+            var requestSettings = apiSettings ?? PlayFabSettings.staticSettings;
```

Comment:
>requestSettings [](start = 16, length = 15)

Suggestion: "activeSettings". I was initially confused by the name "requestSettings", assuming that it's the settings that a user specified in his/her request object